presto-common/src/main/java/com/facebook/presto/common/type/TypeSignature.java

```diff
@@ -273,13 +281,14 @@ VarcharEnumMap getVarcharEnumMap()
     private static EnumMapParsingData parseEnumMap(String signature, int startIndex)
     {
         EnumMapParsingState state = EnumMapParsingState.EXPECT_KEY;
-        boolean isLongEnum = signature.substring(startIndex).trim().toLowerCase(ENGLISH).startsWith(LONG_ENUM_PREFIX);
-        int openBracketIndex = startIndex + (isLongEnum ? LONG_ENUM_PREFIX.length() : VARCHAR_ENUM_PREFIX.length()) + 1;
+        boolean isBigintEnum = signature.startsWith(BIGINT_ENUM_PREFIX, startIndex - BIGINT_ENUM_PREFIX.length() - 1);
```

Comment:
Is the signature already case normalized?

Reply:
Yes, the caller normalized the signature before calling. There are tests cases to test this as well.